interfaces/builtin/shared_memory.go

```diff
@@ -321,7 +326,10 @@ func (iface *sharedMemoryInterface) AppArmorConnectedSlot(spec *apparmor.Specifi
 }
 
 func (iface *sharedMemoryInterface) MountConnectedPlug(spec *mount.Specification, plug *interfaces.ConnectedPlug, slot *interfaces.ConnectedSlot) error {
-	if !iface.isPrivate(plug) {
+	private, err := iface.isPrivate(plug)
+	if err != nil {
+		return err
+	} else if !private {
```

Comment:
(very nitpick) I'm personally not a huge fan of using `else if` after an `if` statement that always returns -- I prefer a new separate `if` for the next condition that might require a return, so the error handling case is more semantically separated from the `!private`, which is not an error case. But the latter is more verbose, and this is really just personal preference on my part.

Reply:
Fwiw @olivercalder, the "effective go" section about "if" https://go.dev/doc/effective_go#if has a similar recommendation/preference

Reply:
That's a good suggestion, thanks! I've split it now